From 9c65206dea76d3b09b7eca4ffcdb6f469e6e7bf2 Mon Sep 17 00:00:00 2001
From: Richard Tollerton <rich.tollerton@ni.com>
Date: Thu, 25 Mar 2021 18:35:49 -0500
Subject: [PATCH 24/25] Revert "rename: a new file for Curl_rename()"

This reverts commit 330f133224af18c65b9325d9b6502e07b4f09f6b.

Uses Curl_rename(), which presently is implemented on Win32 with MoveFileEx(),
which does not exist on Phar Lap.

Upstream-Status: Inappropriate [wacky builds]
Signed-off-by: Richard Tollerton <rich.tollerton@ni.com>
---
 lib/Makefile.inc |  2 --
 lib/cookie.c     | 28 +++++++++++++++++--
 lib/rename.c     | 71 ------------------------------------------------
 lib/rename.h     | 27 ------------------
 4 files changed, 26 insertions(+), 102 deletions(-)
 delete mode 100644 lib/rename.c
 delete mode 100644 lib/rename.h

diff --git a/lib/Makefile.inc b/lib/Makefile.inc
index e8d2259..9f3989f 100644
--- a/lib/Makefile.inc
+++ b/lib/Makefile.inc
@@ -175,7 +175,6 @@ LIB_CFILES =         \
   progress.c         \
   psl.c              \
   rand.c             \
-  rename.c           \
   rtsp.c             \
   select.c           \
   sendf.c            \
@@ -295,7 +294,6 @@ LIB_HFILES =         \
   psl.h              \
   quic.h             \
   rand.h             \
-  rename.h           \
   rtsp.h             \
   select.h           \
   sendf.h            \
diff --git a/lib/cookie.c b/lib/cookie.c
index 09fd092..203c4a5 100644
--- a/lib/cookie.c
+++ b/lib/cookie.c
@@ -98,7 +98,6 @@ Example set of cookies:
 #include "inet_pton.h"
 #include "parsedate.h"
 #include "rand.h"
-#include "rename.h"
 
 /* The last 3 #include files should be in this order */
 #include "curl_printf.h"
@@ -1503,6 +1502,31 @@ static char *get_netscape_format(const struct Cookie *co)
     co->value?co->value:"");
 }
 
+/* return 0 on success, 1 on error */
+static int xrename(const char *oldpath, const char *newpath)
+{
+#ifdef WIN32
+  /* rename() on Windows doesn't overwrite, so we can't use it here.
+     MoveFileExA() will overwrite and is usually atomic, however it fails
+     when there are open handles to the file. */
+  const int max_wait_ms = 1000;
+  struct curltime start = Curl_now();
+  for(;;) {
+    timediff_t diff;
+    if(MoveFileExA(oldpath, newpath, MOVEFILE_REPLACE_EXISTING))
+      break;
+    diff = Curl_timediff(Curl_now(), start);
+    if(diff < 0 || diff > max_wait_ms)
+      return 1;
+    Sleep(1);
+  }
+#else
+  if(rename(oldpath, newpath))
+    return 1;
+#endif
+  return 0;
+}
+
 /*
  * cookie_output()
  *
@@ -1590,7 +1614,7 @@ static int cookie_output(struct Curl_easy *data,
   if(!use_stdout) {
     fclose(out);
     out = NULL;
-    if(Curl_rename(tempstore, filename)) {
+    if(xrename(tempstore, filename)) {
       unlink(tempstore);
       goto error;
     }
diff --git a/lib/rename.c b/lib/rename.c
deleted file mode 100644
index f858d43..0000000
--- a/lib/rename.c
+++ /dev/null
@@ -1,71 +0,0 @@
-/***************************************************************************
- *                                  _   _ ____  _
- *  Project                     ___| | | |  _ \| |
- *                             / __| | | | |_) | |
- *                            | (__| |_| |  _ <| |___
- *                             \___|\___/|_| \_\_____|
- *
- * Copyright (C) 2020, Daniel Stenberg, <daniel@haxx.se>, et al.
- *
- * This software is licensed as described in the file COPYING, which
- * you should have received as part of this distribution. The terms
- * are also available at https://curl.se/docs/copyright.html.
- *
- * You may opt to use, copy, modify, merge, publish, distribute and/or sell
- * copies of the Software, and permit persons to whom the Software is
- * furnished to do so, under the terms of the COPYING file.
- *
- * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
- * KIND, either express or implied.
- *
- ***************************************************************************/
-
-#include "rename.h"
-
-#include "curl_setup.h"
-
-#if (!defined(CURL_DISABLE_HTTP) || !defined(CURL_DISABLE_COOKIES)) || \
-  !defined(CURL_DISABLE_ALTSVC)
-
-#include "curl_multibyte.h"
-#include "timeval.h"
-
-/* The last 3 #include files should be in this order */
-#include "curl_printf.h"
-#include "curl_memory.h"
-#include "memdebug.h"
-
-/* return 0 on success, 1 on error */
-int Curl_rename(const char *oldpath, const char *newpath)
-{
-#ifdef WIN32
-  /* rename() on Windows doesn't overwrite, so we can't use it here.
-     MoveFileEx() will overwrite and is usually atomic, however it fails
-     when there are open handles to the file. */
-  const int max_wait_ms = 1000;
-  struct curltime start = Curl_now();
-  TCHAR *tchar_oldpath = curlx_convert_UTF8_to_tchar((char *)oldpath);
-  TCHAR *tchar_newpath = curlx_convert_UTF8_to_tchar((char *)newpath);
-  for(;;) {
-    timediff_t diff;
-    if(MoveFileEx(tchar_oldpath, tchar_newpath, MOVEFILE_REPLACE_EXISTING)) {
-      curlx_unicodefree(tchar_oldpath);
-      curlx_unicodefree(tchar_newpath);
-      break;
-    }
-    diff = Curl_timediff(Curl_now(), start);
-    if(diff < 0 || diff > max_wait_ms) {
-      curlx_unicodefree(tchar_oldpath);
-      curlx_unicodefree(tchar_newpath);
-      return 1;
-    }
-    Sleep(1);
-  }
-#else
-  if(rename(oldpath, newpath))
-    return 1;
-#endif
-  return 0;
-}
-
-#endif
diff --git a/lib/rename.h b/lib/rename.h
deleted file mode 100644
index 534f747..0000000
--- a/lib/rename.h
+++ /dev/null
@@ -1,27 +0,0 @@
-#ifndef HEADER_CURL_RENAME_H
-#define HEADER_CURL_RENAME_H
-/***************************************************************************
- *                                  _   _ ____  _
- *  Project                     ___| | | |  _ \| |
- *                             / __| | | | |_) | |
- *                            | (__| |_| |  _ <| |___
- *                             \___|\___/|_| \_\_____|
- *
- * Copyright (C) 2020, Daniel Stenberg, <daniel@haxx.se>, et al.
- *
- * This software is licensed as described in the file COPYING, which
- * you should have received as part of this distribution. The terms
- * are also available at https://curl.se/docs/copyright.html.
- *
- * You may opt to use, copy, modify, merge, publish, distribute and/or sell
- * copies of the Software, and permit persons to whom the Software is
- * furnished to do so, under the terms of the COPYING file.
- *
- * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
- * KIND, either express or implied.
- *
- ***************************************************************************/
-
-int Curl_rename(const char *oldpath, const char *newpath);
-
-#endif /* HEADER_CURL_RENAME_H */
-- 
2.25.1

